src/AdoNetCore.AseClient/AseErrorCollection.cs

```diff
@@ -10,13 +11,11 @@ public sealed class AseErrorCollection : ICollection
     {
         private readonly object _syncRoot = new object();
         private readonly AseError[] _errors;
-        private readonly int _indexOfMostSevereError;
-        public AseError MainError { get { return _indexOfMostSevereError == -1 ? null : _errors[_indexOfMostSevereError]; } }
+        internal AseError MainError => _errors.Length > 0 ? _errors[0] : null;
 
-        internal AseErrorCollection(params AseError[] errors) 
+        internal AseErrorCollection(params AseError[] errors)
         {
-            _errors = errors ?? new AseError[0];
-            _indexOfMostSevereError = GetIndexOfMostSevereError();
+            _errors = new List<AseError>(ArrangeErrors(errors)).ToArray();
```

Comment:
I think this traverses the collection twice. Once in the List<T> constructor, and again in ToArray. Probably a more efficient way to do this.